lib/ansible/plugins/filter/network.py

```diff
@@ -0,0 +1,112 @@
+#
+# {c) 2017 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import urllib2
+import urlparse
+
+try:
+    import textfsm
+    HAS_TEXTFSM = True
+except ImportError:
+    HAS_TEXTFSM = False
+
+from ansible.module_utils.six import StringIO
+from ansible.module_utils.netcfg import NetworkConfig, dumps
+from ansible.errors import AnsibleError
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+NETWORK_OS = {
+    'ios': 1,
+    'iosxr': 1,
+    'nxos': 2,
+    'eos': 4
+}
+
+def diff_network_config(value, base, network_os=None, indent=1, match='line', replace='line'):
+
+    assert match in ('line', 'strict', 'exact', 'none')
+    assert replace in ('line', 'block')
+    assert network_os in NETWORK_OS.keys()
+
+    if network_os:
```

Comment:
Not sure how this would ever hit base on the previous assert? Perhaps the assert should be here?